### PR TITLE
integrating generate renders to pipeline

### PR DESCRIPTION
**Pull request by Arshitha**
This PR includes the following changes:

- visualqc t1_mri prep code removed
- integrated generate renders function into the defacing scripts
- some light code refactoring to consistently use pathlib library 

I've tested the code on two sample datasets from start to end, and its working as expected.

Things to do before it can be merged:
- README edits

### CODE REVIEW

**Comment by Arshitha** [comment]
README has no been updated.

File: src/run.py [hunk 1]
```diff
@@ -60,37 +59,46 @@ def get_sess_dirs(subj_dir_path, mapping_dict):
...
+    output = args.output_dir.resolve()
     no_clean = args.no_clean
 
     to_deface = []
     # for one subject or list of subjects (and all their sessions, if present)
-    if args.participant_label != None:
+    if args.participant_label is not None:
         for p in args.participant_label:
-            to_deface.extend(glob(os.path.join(args.bids_dir, f'sub-{p}', "ses-*")))
+            to_deface.extend(list(args.bids_dir.joinpath(f'sub-{p}').glob("ses-*")))
```
Comment (ericearl) [inline src/run.py:84]: glob 1. These comments are just making sure this was intentional.

File: src/run.py [hunk 2]
```diff
@@ -60,37 +59,46 @@ def get_sess_dirs(subj_dir_path, mapping_dict):
...
+
             if not to_deface:
-                to_deface = glob(os.path.join(args.bids_dir, f'sub-{p}'))
+                to_deface = list(args.bids_dir.glob(f'sub-{p}'))
```
Comment (ericearl) [inline src/run.py:87]: glob 2. These comments are just making sure this was intentional.

File: src/run.py [hunk 3]
```diff
@@ -60,37 +59,46 @@ def get_sess_dirs(subj_dir_path, mapping_dict):
...
 
     # only for one subset of sessions
-    if args.session_id != None:
+    if args.session_id is not None:
         for s in args.session_id:
-            to_deface = glob(os.path.join(args.bids_dir, "sub-*", f'ses-{s}'))
+            to_deface = list(args.bids_dir.rglob(f'ses-{s}'))
```
Comment (ericearl) [inline src/run.py:92]: rglob 1. These comments are just making sure this was intentional.

File: src/run.py [hunk 4]
```diff
@@ -60,37 +59,46 @@ def get_sess_dirs(subj_dir_path, mapping_dict):
...
     # for all sessions
-    if args.participant_label == None and args.session_id == None:
-        session_check = glob(os.path.join(args.bids_dir, "sub-*", "ses-*"))
+    if args.participant_label is None and args.session_id is None:
+        session_check = list(args.bids_dir.rglob("ses-*"))
```
Comment (ericearl) [inline src/run.py:95]: rglob 2. These comments are just making sure this was intentional.

File: README.md [hunk 5]
```diff
@@ -46,12 +46,12 @@ Once conda finishes creating the virtual environment, activate `dsstdeface`.
...
 ```
 
-## Using `dsst_defacing_wf.py`
+## Usage
 
 To deface anatomical scans in the dataset, run the `src/dsst_defacing_wf.py` script. From within the `dsst-defacing-pipeline` cloned directory, run the following command to see the help message.
 
 ```text
-% python src/dsst_defacing_wf.py -h
+% python src/run.py -h
```
Comment (ericearl) [inline README.md:54]: incredibly minor piece of feedback, but for more obviousness, I would be inclined to put the run.py at the root of the repository folder structure in the parent of `src/`. Not necessary to accept this PR.

Comment (ericearl) [reply README.md:54]: Option 1 and Option 2 below still have `src/dsst_defacing_wf.py` instead of `src/run.py`.

Comment (ericearl) [reply README.md:54]: It was a quick fix so I fixed and committed it to this branch

**Review by ericearl — DISMISSED** [review]
I approve (and I'm testing right now and it looks like it won't have any problems)!

**Comment by ericearl** [comment]
I spoke too soon! I got a lot of:

`Has the render been created? False`

I was testing over an SSH connection on curium from my VPN'ed PC. I don't know whether or not that should have worked... I will try direct on curium one more time and then I'm done. But this begs the question, "how can a user ONLY generate renders if they fail the first time and save all that defacing again?" we may want to file an issue to add `--only-generate-renders` to skip the defacing over again.

File: src/run.py (hunk 1 above)
Comment (Arshitha) [reply src/run.py:84]: Yes, this was intentional just to stay consistent with usage of pathlib library coz all the other scripts use pathlib.Path objects. And I think it's a bit more readable too.

File: README.md (hunk 5 above)
Comment (Arshitha) [reply README.md:54]: thanks for catching those!

File: src/run.py (hunk 3 above)
Comment (Arshitha) [reply src/run.py:92]: yes, and same reasoning as above.

**Comment by Arshitha** [comment]
@ericearl I'll try this out on curium too but just to confirm, did you have fsleyes on your env?

**Comment by ericearl** [comment]
It was the conda environment, got me again.

File: README.md [hunk 6]
```diff
@@ -122,60 +106,10 @@ If you have dataset with over 10 subjects and since each defacing job is indepen
 subject/session in the dataset using the `-n/--n-cpus` option. The following example command will run the pipeline occupying 10 processors at a time.
 
 ```bash
-python src/dsst_defacing_wf.py ${INPUT_DIR} ${OUTPUT_DIR} -n 10
+python src/run.py ${INPUT_DIR} ${OUTPUT_DIR} -n 10
```
Comment (ericearl) [inline README.md:109]: please make this a 5-line code block where the first line is conda activate. something like:

```suggestion
# activate your conda environment
conda activate

# once your conda environment is active, execute the following
python src/run.py ${INPUT_DIR} ${OUTPUT_DIR} -n 10
```

**Review by ericearl — APPROVED** [review]
Please review my two new README suggestions. Otherwise, I approve this PR.

**Comment by Arshitha** [comment]
@ericearl Great, thank you! I'll merge this to main and do some more testing with the main branch.

**Pull request merged**